pkg/cookies/csrf.go

```diff
@@ -214,16 +211,11 @@ func csrfCookieName(opts *options.Cookie, stateSubstring string) string {
 }
 
 // ExtractStateSubstring extract the initial state characters, to add it to the CSRF cookie name
-func ExtractStateSubstring(req *http.Request) string {
+func ExtractStateSubstring(state string) string {
 	lastChar := csrfStateLength - 1
 	stateSubstring := ""
-
-	state := req.URL.Query()["state"]
-	if state[0] != "" {
```

Comment:
Why not just add an extra conditional here to make sure the state has length more than 0?

Reply:
If state is not received, then CSRF cookie will not be found. I'm trying to get the state first and fail, if it does not exist. Then fail, if cookie does not exist.

Just a check, in the end will result in a CSRF cookie missing, but what is really missing is the state.